cocoa/src/toga_cocoa/app.py

```diff
@@ -379,12 +379,20 @@ def enter_full_screen(self, windows):
         )
 
         for window, screen in zip(windows, NSScreen.screens):
-            window.content._impl.native.enterFullScreenMode(screen, withOptions=opts)
+            # The widgets are actually added to window._impl.container.native, instead of
+            # window.content._impl.native. And window._impl.native.contentView is
+            # window._impl.container.native. But, directly using window._impl.container.native
+            # would be more implementation specific and likely to break with any future
+            # implementation changes. Hence, use  window._impl.native.contentView, which is
+            # guaranteed to be present and assigned on a NSWindow.
+            #
+            # Hence, we need to go fullscreen on window._impl.native.contentView instead.
+            window._impl.content_view.enterFullScreenMode(screen, withOptions=opts)
```

Comment:
As a bonus effect of using `window._impl.native.contentView`, we can lift the requirement of needing a content to be assigned to a window in cocoa, for entering into PRESENTATION mode in #2473